### PR TITLE
update to kirkstone

### PR DESCRIPTION
**Pull request by dremsol**

### CODE REVIEW

**Comment by dremsol** [comment]
Build still fails due to the error in enclosed [log.do_resin_boot_dirgen_and_deploy](https://github.com/balena-os/balena-up-board/files/10838889/log.do_resin_boot_dirgen_and_deploy.txt). @floion could you the build a go?

File: layers/meta-balena-up-board/recipes-kernel/linux/linux-intel_%.bbappend [hunk 1]
```diff
@@ -1,9 +1,8 @@
-inherit kernel-resin
+inherit kernel-balena
 
 FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
 
-SRC_URI += "file://0001-Revert-random-fix-crng_ready-test.patch \
```
Comment (floion) [inline layers/meta-balena-up-board/recipes-kernel/linux/linux-intel_%.bbappend:5]: this line can actually be deleted and not just commented out

File: layers/meta-balena-up-board/recipes-core/images/balena-image.inc [hunk 2]
```diff
@@ -14,6 +14,3 @@ BALENA_BOOT_PARTITION_FILES = " \
     grub-efi-bootx64.efi:/EFI/BOOT/bootx64.efi \
     grubenv:/EFI/BOOT/grubenv \
     "
-
-# add to rootfs the firmware for Ampak AP6214a and AP6355 Wireless/Bluetooth
-IMAGE_INSTALL:append += " ampak-firmware"
```
Comment (floion) [inline layers/meta-balena-up-board/recipes-core/images/balena-image.inc]: The image we produce supports Up Board, Up Core, Up Squared, Up Core Plus. IT's just 1 OS image so removing the wifi firmware will render it broken for the hw which uses it.
Why remove it in the first place?

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    --dry-run \
+    --remove-build \
+    --development-image yes \
+    --log yes \
+    --shared-downloads /var/cache/yocto/downloads \
+    --shared-sstate /var/cache/yocto/sstate-cache
+
+source layers/poky/oe-init-build-env
+
+MACHINE=up-squared bitbake balena-image-flasher
```
Comment (floion) [inline build.sh]: We do not need this file

File: up-squared.coffee [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	configuration:
+		config:
+			partition:
+				primary: 1
+			path: '/config.json'
+
+	options: [ networkOptions.group ]
+
+	initialization: commonImg.initialization
```
Comment (floion) [inline up-squared.coffee:55]: The Up Board image we have already supports Up Squared too.
The only difference I am aware of are the ACPI tables that need to be used. And we allow to specify which ACPI table needs to be used via the dashboard

File: .gitmodules [hunk 5]
```diff
@@ -18,7 +18,10 @@
 	url = https://github.com/meta-rust/meta-rust.git
 [submodule "layers/meta-up-board"]
 	path = layers/meta-up-board
-	url = https://github.com/AaeonCM/meta-up-board.git
+	url = https://github.com/up-division/meta-up-board
 [submodule "contracts"]
 	path = contracts
 	url = https://github.com/balena-io/contracts.git
+[submodule "layers/meta-balena-up-board/recipes-kernel/pinctrl-upboard"]
```
Comment (floion) [inline .gitmodules:22]: Why is this a submodule? The meta-balena-up-board is a regular directory and not a submodule

File: .gitmodules [hunk 6]
```diff
@@ -18,7 +18,10 @@
 	url = https://github.com/meta-rust/meta-rust.git
 [submodule "layers/meta-up-board"]
 	path = layers/meta-up-board
-	url = https://github.com/AaeonCM/meta-up-board.git
+	url = https://github.com/up-division/meta-up-board
```
Comment (floion) [inline .gitmodules:18]: I have asked AAEON if this is the newer official BSP and I am waiting for a reply from them

Comment (dremsol) [reply .gitmodules:18]: I've allready asked Daniel *** from AAEON in February and his answer is the reason for this PR.

```
We have already released, but changed repository.

From now on please refer always to this one, which now include Kirkstone: https://github.com/up-division/meta-up-board

Please make sure you follow the instructions in the readme carefully and let me know if you find any issue (the release is still in beta).
As the recipe integration with the new up pinctrl driver is not complete, you need to download it manually with git after cloning the meta-up-board repository (it is described in the readme).
```

**Comment by floion** [comment]
@resin-jenkins test this please

File: .gitmodules (hunk 6 above)
Comment (floion) [reply .gitmodules:18]: I'd like a word from them if possible

File: layers/meta-balena-up-board/recipes-core/images/balena-image.inc (hunk 2 above)
Comment (dremsol) [reply layers/meta-balena-up-board/recipes-core/images/balena-image.inc]: Currently causes following build error:
`ERROR: QA Issue: ampak-firmware: Files/directories were installed but not shipped in any package`

File: .gitmodules (hunk 6 above)
Comment (dremsol) [reply .gitmodules:18]: Sure, I can forward you the email conversation if you like.

File: layers/meta-balena-up-board/recipes-core/images/balena-image.inc (hunk 2 above)
Comment (dremsol) [reply layers/meta-balena-up-board/recipes-core/images/balena-image.inc]: Seems to be caused by deprecated override syntax upstream [balena-up-board](https://github.com/AaeonCM/meta-up-board/blob/5573c004c3f3bd31c3e5520861730f0d144d2b2d/recipes-firmware/ampak-firmware/ampak-firmware_1.2.bb#L12)

Comment (dremsol) [reply layers/meta-balena-up-board/recipes-core/images/balena-image.inc]: fixed upstream

**Comment by dremsol** [comment]
bumped into a remaining error in kernel-headers-test-1.0-r0 do_compile which i could use some help with.
[log.do_compile.549113.txt](https://github.com/balena-os/balena-up-board/files/11773330/log.do_compile.549113.txt)

**Comment by floion** [comment]
The kernel headers test issue should be solved rather soon since we have some changes addressing that in meta-balena: https://github.com/balena-os/meta-balena/pull/3159

In the meanwhile, can you rebase this PR on top of master please?

**Comment by dremsol** [comment]
> The kernel headers test issue should be solved rather soon since we have some changes addressing that in meta-balena: [balena-os/meta-balena#3159](https://github.com/balena-os/meta-balena/pull/3159)
> 
> In the meanwhile, can you rebase this PR on top of master please?

Done, cleaned up history a bit as well.

**Comment by floion** [comment]
> > The kernel headers test issue should be solved rather soon since we have some changes addressing that in meta-balena: [balena-os/meta-balena#3159](https://github.com/balena-os/meta-balena/pull/3159)
> > In the meanwhile, can you rebase this PR on top of master please?
> 
> Done, cleaned up history a bit as well.

Thanks. Few more things to do here if I may ask you.
Can you remore the updating of the meta-balena, balena-yocto-scripts and contracts submodules? These are done automatically. See https://github.com/balena-os/balena-up-board/pull/424 for example

Second, can you squash https://github.com/balena-os/balena-up-board/pull/355/commits/bd9a4121de70fc32c2ae8c807828443b9b7b60ff and https://github.com/balena-os/balena-up-board/pull/355/commits/b63a613ba46638ec162026490d0ce0662ed25b0a into just 1 commit? They both do the same thing, use the kirkstone override syntax

File: up-squared.coffee (hunk 4 above)
Comment (dremsol) [reply up-squared.coffee]: Just out of curiosity, how would that work exactly?

Comment (floion) [reply up-squared.coffee]: You can do that from the `Configuration` tab in dashboard:

```
Define SSDT overlays. Only supported by supervisor versions >= v. Only supported by supervisor versions >= v10.9.2.

```

File: .gitmodules [hunk 7]
```diff
@@ -18,7 +18,7 @@
 	url = https://github.com/meta-rust/meta-rust.git
 [submodule "layers/meta-up-board"]
 	path = layers/meta-up-board
-	url = https://github.com/AaeonCM/meta-up-board.git
+	url = https://github.com/up-division/meta-up-board.git
```
Comment (floion) [inline .gitmodules:18]: Please also leave a note here that the switch to the new BSP repo has been advised by AAEON themselves

**Comment by floion** [comment]
Hi @dremsol can you squash https://github.com/balena-os/balena-up-board/pull/355/commits/4434eeb11dbad3860aee47a8607fd7d7c2a21bdd into https://github.com/balena-os/balena-up-board/pull/355/commits/e083e3779e82b7be2408f1802ef108e8ffe2b429 ? These 2 commits are both needed for the switch to kirkstone

**Comment by floion** [comment]
@resin-jenkins test this please

**Comment by dremsol** [comment]
@floion seems that image size is to big

```bash
ERROR: balena-image-initramfs-1.0-r0 do_image_cpio: The rootfs size 40960(K) exceeds IMAGE_ROOTFS_MAXSIZE: 32768(K)
ERROR: Logfile of failure stored in: /work/build/tmp/work/up_board-poky-linux/balena-image-initramfs/1.0-r0/temp/log.do_image_cpio.101662
NOTE: recipe balena-image-initramfs-1.0-r0: task do_image_cpio: Failed
ERROR: Task (/work/build/../layers/meta-balena/meta-balena-common/recipes-core/images/balena-image-initramfs.bb:do_image_cpio) failed with exit code '1'
```
[log.do_image_cpio.101662.txt](https://github.com/balena-os/balena-up-board/files/11979898/log.do_image_cpio.101662.txt)

**Comment by dremsol** [comment]
@floion did you had the opportunity to analyze the `IMAGE_ROOTFS_MAXSIZE`? Could you maybe point me in the right direction to sort this out?

**Comment by floion** [comment]
@resin-jenkins ok to test

**Comment by floion** [comment]
Hi @dremsol not yet; will look at it

**Comment by floion** [comment]
@dremsol we could enable firmware compression like here https://github.com/balena-os/balena-intel/commit/003be502f54d49a48bfd47391655fc22c7cac77e

**Comment by dremsol** [comment]
@floion that seems to be enabled allready right? [meta-balena-up-board/conf/layer.conf](https://github.com/drebble/balena-up-board/blob/ffc2a024f481ddd5dc3f2241cb5596cdc8004925/layers/meta-balena-up-board/conf/layer.conf#L16)

**Comment by floion** [comment]
Superseded by https://github.com/balena-os/balena-up-board/pull/556

**Pull request closed without merging**